Added an About category with Partners view (#1pv7h)

jerbob opened the pull request:
• Added the "About" category, accessible via the Navbar
• The About category now contains the Server Rules, as well as a new Partners page
• The Partners page contains cards for each partner, with responsive columns

![image](https://user-images.githubusercontent.com/25161069/38618290-1197062a-3d91-11e8-8c28-4f6f0d8c28fb.png)

![image](https://user-images.githubusercontent.com/25161069/38618310-1ffa9c18-3d91-11e8-9c2b-af30c53e06f7.png)

Commit:
Added Partners view, and Navbar item

Commit:
Moved info/rules to about/rules and fixed references

Commit:
Added Partner images, and populated the partners page

Commit:
Fixed responsiveness on smaller displays

Added like 2 characters to the column tag, to enable stacking of columns if the display
width is lower than 960 pixels.

Commit:
Merge branch 'master' into partners

Commit:
Fixed indentation

Commit:
Merge branch 'master' into partners

gdude2002 commented on templates/main/index.html:
```diff
@@ -32,4 +32,4 @@ <h1 class="uk-header">
     </div>
 </div>
-{% endblock %}
+{% endblock %}
```
Missing newline at end of file

gdude2002 commented on templates/main/about/partners.html:
```diff
@@ -0,0 +1,119 @@
+            </article>
+        </div>
+    </div>
+{% endblock %}
```
Missing newline at end of file

jerbob replied:
Fixed

gdude2002 commented on templates/main/about/index.html:
```diff
@@ -0,0 +1,34 @@
+            </article>
+        </div>
+    </div>
+{% endblock %}
```
Missing newline at end of file

jerbob replied:
Fixed

gdude2002 commented on templates/main/about/partners.html:
```diff
@@ -0,0 +1,119 @@
+        <div class="uk-container uk-container-small">
+            <article class="uk-article">
+                <h1 class="uk-article-title hover-title" id="top">
+                    Partners
```
Can we generate this from a JSON file or something? It seems pretty standard and it'd be nice if we could modify it easily later on

jerbob replied:
The Partners cards are now generated from `static/partners.json`.

Commit:
Updated Code Monkeys Banner

Added the shiny new CM Banner, and rearranged the partner cards.

Commit:
Fixed newlines at end of HTML files

Commit:
Partners page updated to generate from JSON file

• The templates/main/about/partners.html file now uses the
static/partners.json file to generate the Partner cards.

• Flexboxes are nice.

lemonsaurus committed:
Merge branch 'master' into partners

lemonsaurus reviewed (dismissed):
Overall the code looks good, but the partners page does not have a clean system for containing the elements, and it creates severe bugs such as this one:

![image](https://user-images.githubusercontent.com/2098517/38745085-def9f15a-3f43-11e8-8a81-7edb571cee6f.png)

The flexboxing needs to be fixed here before we can commit this. The main container should expand to contain the entire card, and it should be split into two divs. Then we should spit out the partners into the right div by using jinja to alternate the display position.

in simpler terms: 

_build a big flexbox, set it to columns. put two boxes in it. loop through the partner list and put all even partners in the left column. loop through it again and put all odd partners in the right column._

Commit:
Rearranged containers on Partners page

I have no idea how flexboxes work :DDD

Commit:
Merge branch 'master' into partners

Commit:
Merge branch 'partners' of https://github.com/discord-python/site into partners

Git shenanigans

Commit:
Updated Partners page to fix styling issues

Commit:
Moved tag styling to the appropriate CSS file

Commit:
Changed code style of stylesheet :D

lemonsaurus commented on static/style.css:
```diff
@@ -125,4 +125,13 @@ Rendered document fixes
+Styling for Partners page cards
+ */
+
+.uk-card {
```
Are we sure we want all objects with this class to have this styling in the future, or should this perhaps have greater specificity?

lemonsaurus commented on templates/main/about/index.html:
```diff
@@ -0,0 +1,34 @@
+                    Learn about the server, find links to our Partners, and more!
+                </p>
+                <p>
+                    <strong>Server Partners</strong> &nbsp;
```
wait what's that nonbreaking space for?

lemonsaurus commented on templates/main/about/partners.html:
```diff
@@ -0,0 +1,48 @@
+                </h1>
+                <div class="uk-column-1-2@m uk-column-divider">
+                    {% if partners is none %}
+                    <div class="uk-alert-danger" uk-alert>
```
I don't agree with this indentation - conditional content should be indented inside of jinja conditionals imo.

lemonsaurus commented on templates/main/about/partners.html:
```diff
@@ -0,0 +1,48 @@
+                    </div>
+                    {% else %}
+                        {% for partner_name, partner_data in partners.items() %}
+                        <div class="uk-inline">
```
also applies to this.

lemonsaurus commented on templates/main/about/partners.html:
```diff
@@ -0,0 +1,48 @@
+                            <div class="uk-card uk-card-hover uk-card-default">
+                                <div class="uk-card-media-top">
+                                    <a href="{{ partner_data['invite'] }}">
+                                        <img src="{{ partner_data['banner'] }}" alt="{{ partner_name }}">
```
unclosed tag. should be `<img src="{{ partner_data['banner'] }}" alt="{{ partner_name }}"/>`

lemonsaurus reviewed (dismissed):
fuckin' change some shit.

Commit:
Addresses Lemon's code style issues

• Partner cards now have id="partner-card", which is selected with a CSS
selector for styling, rather than styling all card objects.
• Removed nonbreaking spaces that PyCharm didn't tell me about
• Indented conditional HTML content to match Jinja conditionals
• Fixed unclosed image tag

lemonsaurus commented on templates/main/about/partners.html:
```diff
@@ -0,0 +1,48 @@
+                        </div>
+                    {% else %}
+                        {% for partner_name, partner_data in partners.items() %}
+                            <div class="uk-inline">
```
I don't think you need this div. You could just apply this class to the card div itself, no? Be wary of div bloat.

lemonsaurus commented on templates/main/about/partners.html:
```diff
@@ -0,0 +1,48 @@
+                    {% else %}
+                        {% for partner_name, partner_data in partners.items() %}
+                            <div class="uk-inline">
+                                <div class="uk-card uk-card-hover uk-card-default" id="partner-card">
```
id's must be unique, so I'm afraid this isn't good enough. You've got the right idea, though. how about you put an ID on the `<div class="uk-column-1-2@m uk-column-divider">`, and use a selector that says "select the divs inside of a div with this id, but only if they have the class `uk-card`. That's very specific so it won't accidentally style someone elses code, and it also won't apply to the uk-alert-danger div.

lemonsaurus reviewed (dismissed):
aaalmost there.

Commit:
Removed div bloating, and increased specifity of CSS selection

lemonsaurus reviewed (approved):
Looks :ok_hand:
